torch/_dynamo/variables/tensor.py

```diff
@@ -283,7 +270,7 @@ def call_method(
                 dim = kwargs.pop("dim")
                 constant_result = constant_result.getitem_const(dim)
 
-        elif name == "size" and self.size is None and config.dynamic_shapes:
+        elif name == "size" and self.size is None:
```

Comment:
There's something a bit odd here though. Does that mean that with dynamic shapes by default, we directly trace size() calls into the graph? If everything in the size call is static, it may be more friendly for the backend to not trace in the size call and inline the values as constants. Wondering if this was hit in tests... maybe not for Inductor, since Inductor can handle it, but possibly for other backends.

Reply:
Let's followup offline, I want to learn more here :)

Reply:
I think you still need to follow up on this. In particular, I suspect this is causing some of the inductor problems which caused you to disable dynamic shapes.

Reply:
Yea, I can check this.